include/lo2s/time/time.hpp

```diff
@@ -94,43 +94,47 @@ class ClockProvider
         {
             "realtime", CLOCK_REALTIME,
         },
-#ifdef _POSIX_MONOTONIC_CLOCK
+#ifdef CLOCK_MONOTONIC
         {
             "monotonic", CLOCK_MONOTONIC,
         },
 #endif
-#ifdef _POSIX_CPUTIME
+#ifdef CLOCK_PROCESS_CPUTIME_ID
         {
             "process-cputime-id", CLOCK_PROCESS_CPUTIME_ID,
         },
 #endif
-#ifdef _POSIX_THREAD_CPUTIME
+#ifdef CLOCK_THREAD_CPUTIME_ID
         {
             "process-thread-id", CLOCK_THREAD_CPUTIME_ID,
         },
 #endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 28)
+#ifdef CLOCK_MONOTONIC_RAW
```

Comment:
I guess, besides the kernel versions, the correct way here would be using http://cmake.org/cmake/help/v3.12/module/CheckSymbolExists.html for each clock not covered by _POSIX macros

Reply:
to my understanding,we can't use CheckSymbolExists here, as it will fail with enums. The best solution here would be using The CheckEnumValueExists check I've written for this PR because it just basically checks if you can assign whatever you are checking to an int.